File: music-manager-web/src/app/models/album.ts
import { Track } from './track';

export class Album {
  id: number;
  name: string;
  imageUri: string;
  summary: string;
  releaseDate: Date;
  numberOfTracks: string;
  tracks: Track[];
  genres: string[];
}

export class AlbumPost {
  id: string;
  name: string;
  artistId: string;
  imageUri: string;
  summary: string;
  releaseDate: Date;
  tracks: Track[];
  genres: string[];
}

File: MusicManager.Web.API/MusicManager.Web.API/Domain/Models/Album.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MusicManager.Web.API.Domain.Models
{
	public class Album
	{
		public int AlbumId { get; set; }
		public int ArtistId { get; set; }
		public int? ImageRefId { get; set; }
		public string Name { get; set; }
		public DateTime ReleaseDate  { get; set; }
		public string Summary { get; set; }
		public DateTime DateAdded { get; set; }
		public DateTime? DateModified { get; set; }
		public DateTime? DateDeleted { get; set; }

		//References
		public Artist Artist { get; set; }
		public List<Track> Tracks { get; set; }
		public ImageRef ImageRef { get; set; }
		public List<AlbumGenre> AlbumGenres { get; set; }
	}
}
